### DISCUSSION TITLE
AntiAlias settings seems to not work with SVG files

### DISCUSSION THREAD

**Discussion by mirko-horvacki** [dlemstra/Magick.NET#1328] [Help] [answered]
Hi,

I tried to rasterize SVG with following code:

`MagickReadSettings settingsVectorCheckSize = new MagickReadSettings();`
`settingsVectorCheckSize.AntiAlias = false;`
`settingsVectorCheckSize.Density = new Density(imageDPi, DensityUnit.PixelsPerInch);`
`settingsVectorCheckSize.ColorSpace = ColorSpace.sRGB;`
`MagickImage imageCheckSize = new MagickImage(byteArray, settingsVectorCheckSize);`
`imageCheckSize.Format = MagickFormat.Png;`

SVG is:
![Original](https://user-images.githubusercontent.com/23114355/210774046-5408fa93-d648-492e-863d-e570b50819e9.svg)

Issue is that setting AntiAlias to false did not prevented rasterization to introduce new colors. 
On letter edges to me it looks like AntiAlias is there and we have shades of red and gray in the image.

Is it possible to somehow rasterize SVG without introducing any new color beside ones defined in SVG? 

For PDF and EPS files, for example, setting AntiAlias to false works, and new colors are not introduced.

Thanks in advance!

Kind regards,

Mirko

**Comment by dlemstra** [comment, marked as answer]
This setting is ignored by the library that is used (librsvg) and I also don't know if it would be possible to add support for that in the future. Maybe turning off the alpha channel would work for you?

**Reply by mirko-horvacki** [reply]
Hm, I think that will not work. Problem is that after rasterization shades of red and gray are not just pixels with different alpha channel values - those are pixels with different RGB values.

**Reply by mirko-horvacki** [reply]
Maybe you meant something else that would involve editing SVG before rasterization?

**Reply by dlemstra** [reply]
I thought you were getting pixels with alpha but it seems you are not so that won't work out. But I just found out that you can reduce the number of colors with this attribute: `<path shape-rendering="crispEdges" `. Output only has three colors and doesn't look nice but maybe this will work for you.
